### PR TITLE
boards: particle_xenon: Add support Xenon

### PR DESCRIPTION
**Pull request by ekarlso**

### CODE REVIEW

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]: Hmmm... Looks like you copied the reel board documentation (as a starting place) but then didn't edit it to become the particle xenon board documentation.  Oops. :)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (ekarlso) [reply boards/arm/particle_xenon/doc/particle_xenon.rst]: It's todo ;) Just needed a quick board up for a friend. I'll fix tomorrow

**Comment by pabigot** [comment]
My boards finally arrived so I'm starting to play with this.

For consistency with other boards based on Nordic hardware, shouldn't it be named something like `nrf52840_xenon`?

**Comment by ekarlso** [comment]
@pabigot I kind of disagree, if you look at the reel board it's named reel_board, also there are 2 other boards we can add namely the argon and boron .

**Comment by pabigot** [comment]
Most other boards like the ble400, sparkfun, and adafruit that use Nordic hardware but are not from Nordic follow the convention; reel_board and 96b_nitrogen are exceptions for nrf52, and there are a couple more for nrf51.  The question is what's the standard (if there is one), and whether a new board does (or should) conform to it.  Nothing rules out `nrf52840_argon`, though I do think `nrf52840_particle_argon` is getting a bit long.

**Comment by carlescufi** [comment]
@ekarlso @pabigot I have no real preference here. We have tended to use `nrf*_` for Nordic Development Kits but there is no real standard behind it.
Maybe @anangl or @ioannisg have a preference?

**Comment by carlescufi** [comment]
@dbkinder can you take another look?
@ekarlso can you fix the compliance issues?

File: boards/arm/particle_xenon/particle_xenon.dts
Comment (pabigot) [inline boards/arm/particle_xenon/particle_xenon.dts]: Add the reset button on P0.18 since that may be configured for application use.

File: boards/arm/particle_xenon/particle_xenon.dts
Comment (pabigot) [inline boards/arm/particle_xenon/particle_xenon.dts]: Add rts-pin and cts-pin, as with pca10056.  (I now understand how the magic works for GPIOs specified as integers: they get passed through directly to the nrfcx hal which knows how to convert the linearized ones to instance-specific psel values.)

File: boards/arm/particle_xenon/board.c
Comment (pabigot) [inline boards/arm/particle_xenon/board.c]: Remove this: it produces an unused variable diagnostic.

**Review by pabigot — CHANGES_REQUESTED** [review]
Minor issues but the board works for me in basic testing (blink, uart output, cdc_acm output).  Thanks for putting this together!

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]:
```suggestion
`particle xenon`_ is a evaluation board based on the Nordic Semiconductor
```

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]:
```suggestion
It is equipped with a onboard LIPO circuit and conforms with
the Adafruit Feather form factor.
```

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]: Can you provide a link/reference for this documentation?

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]:
```suggestion
Build and flash an application in the usual way, for example:
```

File: boards/arm/particle_xenon/doc/particle_xenon.rst
Comment (dbkinder) [inline boards/arm/particle_xenon/doc/particle_xenon.rst]: I'd recommend deleting this heading (and underline)

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by ioannisg — COMMENTED** [review]
Could you also remove board.c, since it looks empty?

**Review by pabigot — CHANGES_REQUESTED** [review]

**Comment by pabigot** [comment]
@ekarlso will you be able to get back to this soon?  There's also PR #11996 pending.

File: boards/common/blackmagic.board.cmake
Comment (pabigot) [inline boards/common/blackmagic.board.cmake]: This change doesn't belong in this PR.

**Comment by pabigot** [comment]
For gitlint the commit message requires a body, not just a subject line and signoff.

For identity/emails: you misspelled your name in your git `user.email` config setting, causing a mismatch.
```
Author: Endre Karlson <endre.karslon@gmail.com>
Date:   Wed Nov 21 22:19:01 2018 +0100

    boards: particle_xenon: Add support Xenon
    
    Signed-off-by: Endre Karlson <endre.karlson@gmail.com>
```

For license some subset (possibly all) of `board.cmake`, `particle_xenon.yaml`, and `particle_xenon_defconfig` need copyright/SPDX headers.

For checkpatch: `particle_xenon.dts` has spaces where it should have tabs.
```
git format-patch -1 --stdout | ${ZEPHYR_BASE}/scripts/checkpatch.pl -
```

**Comment by ekarlso** [comment]
@pabigot but which file is missing license ? I can't see which based upon what other boards have

**Comment by pabigot** [comment]
All three files I listed are missing a copyright and license block.  Note that this check was made more strict after most of the other boards were merged, so that it's missing for them is not relevant: it's needed now.

**Comment by galak** [comment]
@pabigot & @dbkinder can you review again

**Review by pabigot — DISMISSED** [review]

**Review by ioannisg — APPROVED** [review]

File: boards/arm/particle_xenon/Kconfig
Comment (pfalcon) [inline boards/arm/particle_xenon/Kconfig]: This depends on CONFIG_BOARD_PARTICLE_XENON, i.e. clearly depends on a particular board, while introduces an option generally named CONFIG_BOARD_ENABLE_DCDC . This is highly confusing.

Nor I personally think that such an option (even named non-confusingly is needed), given that CONFIG_SOC_DCDC_NRF52X already exists. Just use it, and make your board code do things depending on whether it's set or not.

And if you're concerned how users would discover that option - well, that's what docs are for ;-).

**Review by pfalcon — DISMISSED** [review]

**Review by carlescufi — APPROVED** [review]

File: boards/arm/particle_xenon/Kconfig
Comment (pfalcon) [reply boards/arm/particle_xenon/Kconfig]: Ok, removing my -1, but I the argument that it's confusing still holds ;-).

Comment (pfalcon) [reply boards/arm/particle_xenon/Kconfig]: Confusing and doesn't scale - we can't duplicate each  Zephyr option thru board-specific alias, so there should be no reasons to do that even for some options.

File: boards/arm/particle_xenon/particle_xenon.dts
Comment (anangl) [inline boards/arm/particle_xenon/particle_xenon.dts]: This does not match the documentation on https://docs.particle.io/datasheets/mesh/xenon-datasheet/ (it should be `&gpio0`). Similarly, pins for `spi1` and for `spi0` (CS) are not matching the mentioned documentation. Is this a different board?

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/arm/particle_xenon/particle_xenon.dts
Comment (pabigot) [reply boards/arm/particle_xenon/particle_xenon.dts]: Agree, these should be on `gpio0`.

Also, could you re-order these to the order "led1=red, led2=green, led3=blue" matching the pin assignment order and the common order of RGB?  Or does "GBR" match some Particle API?

Comment (pabigot) [reply boards/arm/particle_xenon/particle_xenon.dts]: @anangl not sure about `spi` CS reference.  P0.31 is `SPI_SS` = "slave select" = CS.  I believe the other SPI pins are also correct.  What are we missing?

File: boards/arm/particle_xenon/particle_xenon.dts
Comment (pabigot) [inline boards/arm/particle_xenon/particle_xenon.dts]: pin numbers are wrong; should be RTS=33, CTS=34 (these are on P1 not P0).

**Review by pabigot — CHANGES_REQUESTED** [review]
Also note out-of-review comment requesting LED order change to RGB/pin-order.

Please coordinate configuration of console device with @locomuco pr #11996.

**Comment by pabigot** [comment]
@ekarlso @locomuco I think it would make sense for there to be a common dts include file for the functional assignment to pins exposed on the Feather header for xenon, argon, and boron.  By eyeball the only difference is that xenon adds a `UART2` which is used internally for connection to a coprocessor on the other two.

It might be easier to get both boards in separately then make them consistent, though, adding base boron at the same time.

**Comment by ekarlso** [comment]
Going ahead and closing this since we now have 1 PR for all Particle boards.

**Pull request closed without merging**
